package/MDAnalysis/analysis/density.py

```diff
@@ -523,32 +523,28 @@ def _set_user_grid(gridcenter, xdim, ydim, zdim, smin, smax):
             Maximum x,y,z coordinates of the user defined grid
     """
     # Check user inputs
-    if (len(gridcenter) != 3) or not isinstance(gridcenter, np.ndarray):
-        raise ValueError('gridcenter is improperly defined')
-    for entry in (xdim, ydim, zdim):
-        if not isinstance(entry, (float, int)):
-            raise ValueError("x/y/z dimensions are improperly defined")
-
-    # Copy gridcenter value to umin and umax
-    umin = np.copy(gridcenter)
-    umax = np.copy(gridcenter)
-
-    # Set min/max of each dimension by shifting by half the edge length
-    # x dimension
-    umin[0] -= xdim/2.0
-    umax[0] += xdim/2.0
-    # y dimension
-    umin[1] -= ydim/2.0
-    umax[1] += ydim/2.0
-    # z dimension
-    umin[2] -= zdim/2.0
-    umax[2] += zdim/2.0
+    try:
+        gridcenter = np.asarray(gridcenter, dtype=np.float32)
+        if gridcenter.shape != (3,):
+            raise ValueError("gridcenter must be a 3D coordinate")
```

Comment:
I think this will not work: the inner `ValueError` will raise the outer one:
```python
In [1]: try:
   ...:     raise ValueError("inner")
   ...: except ValueError:
   ...:     raise ValueError("outer")
   ...:
   ...:
```
gives
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-063acd63d394> in <module>()
      1 try:
----> 2     raise ValueError("inner")
      3 except ValueError:

ValueError: inner

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-1-063acd63d394> in <module>()
      2     raise ValueError("inner")
      3 except ValueError:
----> 4     raise ValueError("outer")

ValueError: outer
```

You need to do the check for `gridcenter.shape != (3,)` outside after the first.

Reply:
Sorry that was a bit of a silly mistake, fixed in b9afd78